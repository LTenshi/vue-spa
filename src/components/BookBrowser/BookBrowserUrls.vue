<template>
  <th>
    <a v-for="link in bookLinks" :key="link.name" :href="link.url" class="bookLink">{{link.name}}<br></a>
  </th>
</template>

<script>
export default {
  props: {
    bookLinks: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style>
.bookLink {
  color: rgb(136, 221, 0);
  text-decoration: none;
}

.bookLink > a:link {
  text-decoration: none;
}

.bookLink > a:visited {
  text-decoration: none;
}

.bookLink > a:hover {
  text-decoration: none;
}

.bookLink > a:active {
  text-decoration: none;
}
</style>