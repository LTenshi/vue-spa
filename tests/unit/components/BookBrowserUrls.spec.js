import BookBrowserUrls from "../../../src/components/BookBrowser/BookBrowserUrls";
import { mount } from "@vue/test-utils";

describe("Book Browser Urls", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(BookBrowserUrls, {
            props: {
               bookLinks: [
                { name: "testing", url: "www.testing.com" },
                { name: "testing 2", url: "www.testing2.com" },
               ]
            }
        });
    })

    it("Should match the snapshot", () => {
        expect(wrapper.element).toMatchSnapshot();
    })

    it('Should contain 2 urls', () => {
        expect(wrapper.html()).toContain("<a href=\"www.testing.com\" class=\"bookLink\">testing<br></a>");
        expect(wrapper.html()).toContain("<a href=\"www.testing2.com\" class=\"bookLink\">testing 2<br></a>");
    });
    
});