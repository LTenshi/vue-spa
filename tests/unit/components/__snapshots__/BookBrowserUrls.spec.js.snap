// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Browser Urls Should match the snapshot 1`] = `
<th>
  
  <a
    class="bookLink"
    href="www.testing.com"
  >
    testing
    <br />
  </a>
  <a
    class="bookLink"
    href="www.testing2.com"
  >
    testing 2
    <br />
  </a>
  
</th>
`;
